core/app/models/spree/order.rb

```diff
@@ -694,6 +694,10 @@ def covered_by_store_credit?
     end
     alias_method :covered_by_store_credit, :covered_by_store_credit?
 
+    def covered_by_valid_payments?
+      return payments.where(state: %w(checkout pending)).sum(:amount) >= total
```

Comment:
Good catch, maybe we should be checking for `checkout`, `pending`, `completed`, and `processing` payments?

It looks like the closest scope on payments is `.valid`, which returns all payments excluding `failed` and `invalid`. It would return voided payments as well, which we probably don't want to consider here. Would it make sense to make a new scope for this?